_sources/Functions/Variablesandparametersarelocal.rst

```diff
@@ -7,117 +7,197 @@
+ particular attention to the variables that are created when the function is called. Note when they are subsequently
+ destroyed as the function returns.
+ 
+ Formal parameters are also local and act like local variables. For example, the lifetime of ``x`` begins when
+ ``square`` is called, and its lifetime ends when the function completes its execution.
+ 
+ So it is not possible for a function to set some local variable to a value, complete its execution, and then when it
+ is called again next time, recover the local variable. Each call of the function creates new local variables, and
+ their lifetimes expire when the function returns to the caller.
+ 
+ **Check Your Understanding**
+ 
+ .. mchoice:: question11_7_1
+    :answer_a: True
+    :answer_b: False
+    :correct: b
+    :feedback_a: Local variables cannot be referenced outside of the function they were defined in.
+    :feedback_b: Local variables cannot be referenced outside of the function they were defined in.
+    :practice: T
+ 
+    True or False: Local variables can be referenced outside of the function they were defined in.
+ 
+ .. fillintheblank:: question11_7_2
+ 
+    Which of the following are local variables? Please, write them in order of what line they are on in the code.
+ 
+    .. sourcecode:: python
+ 
+     numbers = [1, 12, 13, 4]
+     def foo(bar):
+         aug = str(bar) + "street"
+         return aug
+ 
+     addresses = []
+     for item in numbers:
+         addresses.append(foo(item))
+ 
+ 
+    The local variables are
+ 
+    -  :bar: Good work!
+       :aug: While aug is a local variable, it is not the first one in the code.
+       :item: item is not a local variable.
+       :.*: Incorrect, try again.
+    -  :aug: Good work!
+       :bar: While bar is a local variable, it is not the first one in the code.
+       :item: item is not a local variable.
+       :.*: Incorrect, try again.
+ 
+ .. mchoice:: question11_7_3
+    :answer_a: 4
+    :answer_b: 6
+    :answer_c: 10
+    :answer_d: Code will give an error because x and z do not match.
+    :correct: a
+    :feedback_a: Correct, the output is right because the subtract function takes in x as the global variable for the z parameter and puts it into the function. The subtract function uses the local variable y for its return.
+    :feedback_b: Incorrect, look again at what is being produced in the subtract function.
+    :feedback_c: Incorrect, look again at what is being produced in the subtract function.
+    :feedback_d: Incorrect, there shouldn't be any error.
+    :practice: T
+ 
+    What would be the result of running the following code?
+ 
+    .. sourcecode:: python
+ 
+      x = 3 * 2
+      y = 1
+ 
+      def subtract(z):
+          y = 10
+          return y - z
+         
+      print(subtract(x)) 
+ 
+ .. mchoice:: question11_7_4
+    :answer_a: 33
+    :answer_b: 12
+    :answer_c: There is an error in the code.
+    :correct: c
```

Comment:
This code runs fine for me and produces a result of 4 as I expected.

@diamondyucas